### DISCUSSION TITLE
Saving and Switching of Multi-Group Visibility State

### DISCUSSION THREAD

**Discussion by fornewfork** [oxdc/obsidian-vertical-tabs#82] [Ideas]
# Feature Overview

allows users to save and restore the visibility states of multiple Groups

# Use Case

I use different groups to organize my workflow. Depending on different work requirements, I need to flexibly show or hide these Groups.

For example: 
I have 5 Groups: A, B, C, D, and E. 
1. In some cases, I only need to display Groups A, B, and C.
2. In other cases, I only need display Groups C, D, and E.
3. Sometimes, I need to use all 5 Groups simultaneously.

# Current Limitation

Currently, Obsidian's Workspace feature cannot meet this need. While the workspace feature can save window layouts, they cannot maintain the hidden state of Groups. 

For example:
- If the current state is: Group A displayed, Group B displayed, Group C displayed, Group D hidden, Group E hidden
- Save this state as Workspace "Flow 1"
- Later, when reopening Workspace "Flow 1", all Groups A, B, C, D, and E will become displayed.

This behavior limits users' ability to quickly switch Group visibility based on different workflows.

The dirct integration of this plugin with Obsidian's Workspace feature may be complicated.
But Workspace feature is not needed if we can save and restore the visibility states of multiple groups.

# Needed Features

1. **Save Group States**: Allow users to save the current visibility state of all Groups as a named configuration.

2. **Quick Switching**: Provide a dropdown menu ( or command palette option ) for users to quickly switch between different saved configurations.

3. **Edit Configurations**: Allow users to edit saved configurations, such as renaming, deleting, or modifying Group states.

4. **Hotkey Support**: Offer customizable hotkeys for saving the current state and switching to specific configurations.

## Expected Benefits

1. Convenience: Users can quickly switch between different work modes without manually showing/hiding Groups.
2. Reduced Distractions: Help users stay focused by only displaying Groups relevant to the current task.

**Comment by oxdc** [comment]
Thank you! I'm working on this workspace-like features, and I want to push further, not just saving group visibility, but remembering all the customizations, tab history, cursor positions and more. Please stay tuned 🎉
